core/src/main/scala/kafka/coordinator/TransactionLogManager.scala

```diff
@@ -0,0 +1,85 @@
+/**
+  * Licensed to the Apache Software Foundation (ASF) under one or more
+  * contributor license agreements.  See the NOTICE file distributed with
+  * this work for additional information regarding copyright ownership.
+  * The ASF licenses this file to You under the Apache License, Version 2.0
+  * (the "License"); you may not use this file except in compliance with
+  * the License.  You may obtain a copy of the License at
+  *
+  *    http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+
+package kafka.coordinator
+
+import kafka.common.Topic
+import kafka.utils.{Logging, ZkUtils}
+import org.apache.kafka.common.utils.Utils
+
+import scala.collection.mutable
+import java.util.concurrent.locks.ReentrantLock
+
+import kafka.utils.CoreUtils.inLock
+
+
+/**
+  * Transaction log manager is part of the transaction coordinator that manages the transaction log, which is
+  * a special internal topic.
+  */
+object TransactionLogManager {
+
+}
+
+class TransactionLogManager(val brokerId: Int,
```

Comment:
Seems like this class shares some common bookkeeping with `GroupMetadataManager`. Do you think it's worthwhile to extract a superclass?